backend/src/routes/api/namespaces/namespaceUtils.ts

```diff
@@ -47,40 +70,48 @@ export const applyNamespaceChange = async (
     );
   }
 
-  const selfSubjectAccessReview = await checkNamespacePermission(fastify, request, name);
-  if (isK8sStatus(selfSubjectAccessReview)) {
-    throw createCustomError(
-      selfSubjectAccessReview.reason,
-      selfSubjectAccessReview.message,
-      selfSubjectAccessReview.code,
-    );
-  }
-  if (!selfSubjectAccessReview.status.allowed) {
-    fastify.log.error(`Unable to access the namespace, ${selfSubjectAccessReview.status.reason}`);
-    throw createCustomError('Forbidden', "You don't have the access to update the namespace", 403);
-  }
-
   let labels = {};
+  let checkPermissionsFn = null;
   switch (context) {
     case NamespaceApplicationCase.DSG_CREATION:
-      labels = {
-        'opendatahub.io/dashboard': 'true',
-      };
+      {
+        labels = { 'opendatahub.io/dashboard': 'true' };
+        checkPermissionsFn = checkAdminNamespacePermission;
+      }
       break;
     case NamespaceApplicationCase.MODEL_MESH_PROMOTION:
-      labels = {
-        'modelmesh-enabled': 'true',
-      };
+      {
+        labels = { 'modelmesh-enabled': 'true' };
+        checkPermissionsFn = checkEditNamespacePermission;
+      }
       break;
     case NamespaceApplicationCase.KSERVE_PROMOTION:
-      labels = {
-        'modelmesh-enabled': 'false',
-      };
+      {
+        labels = { 'modelmesh-enabled': 'false' };
+        checkPermissionsFn = checkEditNamespacePermission;
+      }
       break;
     default:
       throw createCustomError('Unknown configuration', 'Cannot apply namespace change', 400);
   }
 
+  const selfSubjectAccessReview = await checkPermissionsFn(fastify, request, name);
+  if (isK8sStatus(selfSubjectAccessReview)) {
```

Comment:
It'll fall into the next if-statement after this -- but I am wondering if we want to put a "dev setup" if statement first -- that basically says "`checkPermissionsFn` is null -- you need to set permissions for all actions" -- basically as a "fail immediately" sorta deal with a dev message so the dev knows the new case they added needs to have a check permissions.

Just my two cents. It should fail in some fashion on the `checkPermissionsFn()` call but I dunno how that will look.

Reply:
Ah, fair enough. Just another `throw createCustomError` maybe? It would be impossible to hit with the current code but that's a good point about future switch cases. Alternatively though we could just have it fall back to `checkAdminNamespacePermission` which is the behavior in all cases before this PR? I'm leaning towards your idea.

Reply:
Yeah, like a 500 error "invalid backend state -- dev broken workflow" or something like that. Something annoying and blunt it was a dev setup. It will be easy to search in the codebase and should prevent anything from being poorly configured. 

It should never happen in production, but it won't mask when code is poorly written -- so it gets caught in the PR for the change. Or in our test infra when we have that going nightly or whatever.

Reply:
Pushed with this change as well.